### PR TITLE
Added new KB files

### LINKED ISSUE #271: [KB] Add KB for google/mirror-branch-action
Knowledge Base is missing for google/mirror-branch-action.

### LINKED ISSUE #272: [KB] Add KB for relekang/python-semantic-release
Knowledge Base is missing for relekang/python-semantic-release.

### LINKED ISSUE #290: [KB] Add KB for google-github-actions/auth
Add KB for google-github-actions/auth

### LINKED ISSUE #302: [KB] Add KB for tommykw/pull-request-reviewer-reminder-action
Knowledge Base is missing for tommykw/pull-request-reviewer-reminder-action.

### PR DESCRIPTION
**Pull request by h0x0er**
Fix #290 
Fix #272 
Fix #302 
Fix #271

### CODE REVIEW

File: knowledge-base/google-github-actions/auth/action-security.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+name: 'Authenticate to Google Cloud'
+# NOTE: GITHUB_TOKEN NOT USED
```
Comment (varunsh-coder) [inline knowledge-base/google-github-actions/auth/action-security.yml:2]: If workload identity federation is used, the `id-token: 'write'` needs to be set. Check the examples in the readme file. This is similar to the existing aws authentication action - do check that action-security.yml. You will need to use conditional statements.

File: knowledge-base/google-github-actions/auth/action-security.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+name: Authenticate to Google Cloud
+github-token:
+  environment-variable-name: ACTIONS_ID_TOKEN_REQUEST_TOKEN
+  is-default: false
+  permissions:
+    id-token: write
+    id-token-if: ${{ contains(with, 'token_format') && with['token_format'] == 'id_token' }}
```
Comment (varunsh-coder) [inline knowledge-base/google-github-actions/auth/action-security.yml:7]: @h0x0er I am not sure if `token_format` is the right thing to look for. In some examples, token_format is not set, but `id_token` is set to `write`. https://github.com/google-github-actions/auth#configuring-gcloud

Comment (h0x0er) [reply knowledge-base/google-github-actions/auth/action-security.yml:7]: @varunsh-coder thanks for the review. I reanalyzed it.. I just figure that `workload_identity_provider` is right candidate to check for.. As if you look [here](https://github.com/google-github-actions/auth/blob/48c46e6a594e8bd0b0b4806155ace5cad3aa8e24/src/main.ts#L92), `ACTIONS_ID_TOKEN_REQUEST_TOKEN` token is used only `workload_identity_provider` is set..

Comment (varunsh-coder) [reply knowledge-base/google-github-actions/auth/action-security.yml]: @h0x0er yes, this seems fine. Can you please rebase? For some reason `GITHUB_TOKEN` is also getting rate limited, so I changed to using a PAT...So the test.yml is different now from when you created the fork...

Comment (h0x0er) [reply knowledge-base/google-github-actions/auth/action-security.yml]: @varunsh-coder Will do that..

File: knowledge-base/google-github-actions/auth/action-security.yml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+name: Authenticate to Google Cloud
+github-token:
+  environment-variable-name: ACTIONS_ID_TOKEN_REQUEST_TOKEN
```
Comment (varunsh-coder) [inline knowledge-base/google-github-actions/auth/action-security.yml:3]: This environment variable `ACTIONS_ID_TOKEN_REQUEST_TOKEN ` is not in the examples, and so the code will not find it. I think we need to use same logic (same as token being default) as in https://github.com/step-security/secure-workflows/blob/815e14aba75030a0c717d030077b54b3755c2e1e/knowledge-base/aws-actions/configure-aws-credentials/action-security.yml#L4
We should document it somewhere. For `id-token` looks like you don't have to send the token to the Action.

**Pull request merged**